includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -310,6 +310,67 @@ public function update_intention(
 		return $this->deserialize_intention_object_from_array( $response_array );
 	}
 
+	/**
+	 * Updates an intention without assuming any mandatory params.
+	 *
+	 * @param string  $intention_id              - The ID of the intention to update.
+	 * @param ?int    $amount                    - Amount to charge.
+	 * @param ?string $currency_code             - Currency to charge in.
+	 * @param ?bool   $save_payment_method       - Whether to setup payment intent for future usage.
+	 * @param ?string $customer_id               - Stripe customer to associate payment intent with.
+	 * @param array   $metadata                  - Meta data values to be sent along with payment intent creation.
+	 * @param array   $level3                    - Level 3 data.
+	 * @param ?string $selected_upe_payment_type - The name of the selected UPE payment type or empty string.
+	 * @param ?string $payment_country           - The payment two-letter iso country code or null.
+	 *
+	 * @return WC_Payments_API_Intention
+	 * @throws API_Exception - Exception thrown on intention creation failure.
+	 */
+	public function update_intention_unrestricted(
+		$intention_id,
+		$amount,
+		$currency_code,
+		$save_payment_method = false,
+		$customer_id = '',
+		$metadata = [],
+		$level3 = [],
+		$selected_upe_payment_type = '',
+		$payment_country = null
+	) {
+		$request = [];
+
+		if ( $amount ) {
+			$request['amount'] = $amount;
+		}
+		if ( $currency_code ) {
+			$request['currency'] = $currency_code;
+		}
+		if ( ! empty( $metadata ) ) {
+			$request['metadata'] = $metadata;
+		}
+		if ( ! empty( $level3 ) ) {
+			$request['level3'] = $level3;
+		}
+		if ( '' !== $selected_upe_payment_type ) {
```

Comment:
Yes, that makes sense. Another way to go about it could be to modify the original `update_intention` to support our use case in a non-disruptive way. Something similar was also proposed here 1331-gh-Automattic/woocommerce-payments-server#discussion_r752249034. I'll look into it 👍

Reply:
This change is no longer required as `update_intention_unrestricted` has been replaced by `update_intention_metadata`.